bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceAction.java

```diff
@@ -316,9 +324,22 @@ public FindReplaceAction(ResourceBundle bundle, String prefix, IWorkbenchWindow
 
 	@Override
 	public void run() {
-		if (fTarget == null)
-			return;
+		// only show Overlay on Text-Editors, don't show it on Consoles
+		if (shouldShowModernOverlay() && fWorkbenchPart instanceof StatusTextEditor) {
+			showModernOverlay();
+		} else {
+			if (fTarget == null) {
+				return;
+			}
```

Comment:
I believe this has something to do with the Overlay only being shown in Editors. However, from reading the code, I could not find an "obvious" reason why the fTarget can't be null because of that. the closes I can see is that in the "update"-method, fTarget is set to null if fWorkbenchPart is null and since the type-check required to branch into "showOverlay" requires fWorkbenchPart to be set, ...

I have pulled the fTarget  == null check outside of this branch!

Reply:
I could not see any obvious reasons why. The unit-tests still run perfectly well when I pull this out of the branches, so I believe this was not a deliberate choice.